aten/src/ATen/MemoryOverlap.cpp

```diff
@@ -48,15 +48,13 @@ MemOverlapStatus get_overlap_status(TensorImpl* a, TensorImpl* b) {
   if (!a->is_non_overlapping_and_dense() || !b->is_non_overlapping_and_dense()) {
     return MemOverlapStatus::TOO_HARD;
   }
-  if (!a->has_storage() || !b->has_storage()) {
-    return MemOverlapStatus::NO;
-  }
   // Test for storage equality, rather than pointer equality.
   // This reduces precision, but if people are aliasing the
   // same pointer across multiple storages there are many
   // similar situations (e.g., storage().data() == storage().data()+1)
   // which we will miss.
-  if (a->storage().is_alias_of(b->storage())) {
+  const auto& a_storage = a->storage();
+  if (a_storage && a_storage.is_alias_of(b->storage())) {
```

Comment:
`storage()` and `has_storage()` are virtual calls, so it's worthwhile to avoid the extra calls. For `torch.add(a, b, out=c)` this saves ~800 instructions.

This does assume that `storage()` doesn't throw an error, but I don't think it makes sense to use this function with non-strided tensors anyway. So, I can't see the harm in them failing.